subprojects/docs/src/docs/userguide/five_things.adoc

```diff
@@ -0,0 +1,125 @@
+
+ 1. Initialization
+ 2. Configuration
+ 3. Execution
+
+In well-designed build scripts, the vast majority of what you see is evaluated during the configuration phase. Even so, some elements are likely to be evaluated during the execution phase, particularly the contents of `doLast {}` and `doFirst {}` blocks.
+
+NOTE: Gradle's build phases are not like Maven's phases. Maven uses its phases to divide the build execution into multiple stages. They serve a similar role to Gradle's task graph, although less flexibly.
+
+When you are authoring or modifying a build, it's important to recognize which bits are evaluated during which phases. Otherwise you will end up with a build that doesn't work as expected. For example, if you set a property within the execution phase, any part of the build that reads that property during the configuration phase won't see the new value.
+
+Consider the following (contrived) example that attempts to initialise a `Copy` task from a properties file. You might try this if the properties file is generated by another task. The problem is that `into` must be set during the configuration phase, otherwise the task won't run. Setting the value within a `doFirst {}` block fails as it happens too late.
+
+[source,groovy]
+.Configuring a task within an action (execution phase) won't work
+----
+task copyFiles(type: Copy) {
+    doFirst {
+        into new Properties().load("my-props.properties").     // <1>
+                getProperty("notes.output.dir")                // <1>
+    }
+    from file("notes")                                         // <2>
+    include "**/*.txt"
+}
+----
+<1> Evaluated during _execution_ phase
+<2> Evaluated during _configuration_ phase
+
+This used to be more of a problem when the `<<` syntax for defining a task was more common, as it implicitly declared a task action — something that is evaluated/run during the execution phase. Users would often attempt to configure the task within that definition and wonder why it wasn't working.
+
+As a general rule of thumb, everything in a build script happens during the configuration phase unless it defines a task action. There are some other exceptions, such as link:{javadocPath}/org/gradle/api/execution/TaskExecutionGraph.html#whenReady-org.gradle.api.Action-[TaskExecutionGraph.whenReady(Action)], but the documentation is clear on their behavior.
+
+You can learn more in the <<build_lifecycle#build_lifecycle,Build Lifecycle>> chapter.
+
+== 4. Gradle is extensible in more ways than one
+
+It would be great if you could build your project using only the build logic bundled with Gradle, but that's rarely possible. Most builds have some special requirements that mean you need to add custom build logic.
+
+Gradle provides several mechanisms that allow you to extend it, from the very simple to the very powerful:
+
+ 1. Extra properties on <<writing_build_scripts#sec:extra_properties,projects>> and <<writing_build_scripts#sec:extra_task_properties,tasks>>.
```

Comment:
(extra) properties are handy but they are untyped and it's hard to know where they come from. Most often there're better ways to do what they are used for, with more modeling.

I would not put them first on that enumeration.

Reply:
The order is from simplest to most complex/work involved. I'm not that keen on changing the order, but as @eriwen suggests, perhaps your concerns can be addressed a little with unnumbered bullets. I can see the numbering implying some sense of priority over and above the item being first in the list.

Reply:
Sounds good 👍

Reply:
Wouldn't it make more sense to order by how common they are and start with custom tasks?

Reply:
Maybe. I don't have any insight into what's most common though. It's probably worth ordering them in such a way as our preferred options come first. I could be persuaded either way.

Reply:
I changed the order to something closer to most common -> less common.